spec-tests/src/test/java/org/candlepin/spec/owners/OwnerResourceSpecTest.java

```diff
@@ -792,4 +797,75 @@ public void shouldAllowUserWithOwnerPoolsPermissionCanSeeSystemPurposeOfTheOwner
     }
 
+    @Test
+    @OnlyInHosted
+    public void shouldMigrateAllConsumersOfAnonymousOwnerToTargetOwner() {
+        String accountId = StringUtil.random("cloud-account-id-");
+        String instanceId = StringUtil.random("cloud-instance-id-");
+        String offerId = StringUtil.random("cloud-offer-");
+        OwnerDTO anonOwner = owners.createOwner(Owners.randomSca().anonymous(true));
+        OwnerDTO destOwner = owners.createOwner(Owners.randomSca());
+        admin.hosted().createOwner(anonOwner);
+        ProductDTO prod = admin.ownerProducts().createProduct(anonOwner.getKey(), Products.random());
+        admin.hosted().createProduct(prod);
+        owners.createPool(anonOwner.getKey(), Pools.random(prod));
+        admin.hosted().createSubscription(Subscriptions.random(anonOwner, prod));
+
+        admin.hosted().associateProductIdsToCloudOffer(offerId, List.of(prod.getId()));
+        admin.hosted().associateOwnerToCloudAccount(accountId, anonOwner.getKey());
+        CloudAuthenticationResultDTO result = ApiClients.noAuth().cloudAuthorization()
+            .cloudAuthorizeV2(accountId, instanceId, offerId, "test-type", "");
+
+        ApiClient anonClient = ApiClients.bearerToken(result.getToken());
+        List<ConsumerDTO> consumers = List.of(
+            anonClient.consumers().createConsumer(Consumers.random(anonOwner)),
+            anonClient.consumers().createConsumer(Consumers.random(anonOwner)),
+            anonClient.consumers().createConsumer(Consumers.random(anonOwner))
+        );
+        List<CertificateDTO> oldIdCerts = consumers.stream().map(ConsumerDTO::getIdCert).toList();
+        List<CertificateDTO> oldScaCerts = fetchCertsOf(consumers, admin);
+
+        AsyncJobStatusDTO job = owners.claim(anonOwner.getKey(),
+            new ClaimantOwner().claimantOwnerKey(destOwner.getKey()));
+
+        job = admin.jobs().waitForJob(job);
+        assertThatJob(job).isFinished();
+
+        // Anon owner should be left with no consumers
+        List<ConsumerDTOArrayElement> anonConsumers = owners.listOwnerConsumers(anonOwner.getKey());
+        assertThat(anonConsumers).isEmpty();
+
+        // All consumers should be migrated to destination owner
+        List<ConsumerDTOArrayElement> migratedConsumers = owners.listOwnerConsumers(destOwner.getKey());
+        assertThat(migratedConsumers).hasSize(3);
+
+        // Anon owner should be claimed
+        OwnerDTO updatedOwner = owners.getOwner(anonOwner.getKey());
+        assertThat(updatedOwner)
+            .returns(true, OwnerDTO::getClaimed)
+            .returns(destOwner.getKey(), OwnerDTO::getClaimantOwner);
+
+        // Consumers should have regenerated identity certificates
+        List<ConsumerDTO> updatedConsumers = fetchConsumers(consumers, admin);
+        List<CertificateDTO> newIdCerts = updatedConsumers.stream().map(ConsumerDTO::getIdCert).toList();
+        assertThat(newIdCerts).doesNotContainAnyElementsOf(oldIdCerts);
```

Comment:
this kind of assertion will wrongly pass if newIdCerts was empty in the first place (for whatever reason). can we make sure both lists actually contain 3 certs?

Reply:
and same thing for sca certs